R/XcmsExperiment-functions.R

```diff
@@ -542,6 +542,53 @@
     res[!is.na(res[, "maxo"]), , drop = FALSE]
 }
 
+
+#' Calculates quality metrics for a chromatographic peak. 
+#'
+#' @param x `list` of peak matrices (from a single MS level and from a single
+#'     file/sample).
+#'
+#' @param rt retention time for each peak matrix.
+#'
+#' @param peakArea `matrix` defining the chrom peak area.
+#'
+#' @author Pablo Vangeenderhuysen
+#'
+#' @noRd
+.chrom_peak_beta_metrics <- function(x, rt, peakArea,
+                                           sampleIndex = integer(),
+                                           cn = character(), ...) {
+  res <- matrix(NA_real_, ncol = 2L, nrow = nrow(peakArea))
+  rownames(res) <- rownames(peakArea)
+  colnames(res) <- c("beta_cor","beta_snr")
+  for (i in seq_len(nrow(res))) {
+    rtr <- peakArea[i, c("rtmin", "rtmax")]
+    keep <- which(MsCoreUtils::between(rt, rtr))
+    if (length(keep)) {
+      xsub <- lapply(x[keep], .pmat_filter_mz,
+                     mzr = peakArea[i, c("mzmin", "mzmax")])
+      ## length of xsub is the number of spectra, the number of peaks can
```

Comment:
I would suggest to remove this comment (lines 570-572). Also I would remove the `mat` calculation (line 573; we will not need that) and the `if (nrow(mat))` (line 574 and 582). The `get_beta_values()` function checks internally that enough data points are provided (minimum is 5) and returns `NA` if not. So we don't need to run this part only if `nrow(mat)` is larger than one - and it saves us additional (unneeded calculations). Also, please remove the `rts <- rep(rt[keep], nr)` line - we don't use the `rts` variable, so also here, no need to calculate that. So, to summarize, simply keep lines 578:580.